### PR TITLE
Add unit-tests for rtp_api serialization and deserialization

### PR DESCRIPTION
**Pull request by t2-g**
*Description of changes:*
This PR adds unit-tests for rtp_api.c serialization and deserialization.

*Test Steps:*

```
git submodule update --init --recursive --checkout test/CMock
cmake -S test/unit-test -B build/ -G "Unix Makefiles" -DCMAKE_BUILD_TYPE=Debug -DBUILD_CLONE_SUBMODULES=ON -DCMAKE_C_FLAGS='--coverage -Wall -Wextra -Werror -DNDEBUG -DLIBRARY_LOG_LEVEL=LOG_DEBUG'
make -C build all
cd build
ctest -E system --output-on-failure
make coverage
```
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/unit-test/rtp_api/rtp_api_utest.c [hunk 1]
```diff
@@ -0,0 +1,892 @@
...
+
+/**
+ * @brief Validate RTP_Serialize in case of valid inputs.
+ */
+void test_Rtp_Serialize_Pass( void )
+{
+    RtpResult_t result;
+    RtpContext_t ctx = { 0 };
+    RtpPacket_t packet = { 0 };
+    uint8_t buffer[ MAX_FRAME_LENGTH ] = { 0 };
```
Comment (ActoryOu) [inline test/unit-test/rtp_api/rtp_api_utest.c]: buffer is declared both globally and locally. Could you name it based on its usage?

File: test/unit-test/rtp_api/rtp_api_utest.c [hunk 2]
```diff
@@ -0,0 +1,892 @@
...
+
+    TEST_ASSERT_EQUAL( RTP_RESULT_OK,
+                       result );
+
+    TEST_ASSERT_EQUAL( 23,
+                       length );
+
+    TEST_ASSERT_EQUAL_UINT8_ARRAY( ( uint8_t * ) "\x80\x60\x04\xD2\x12\x34\x56\x78\x87\x65\x43\x21\x68\x65\x6C\x6C\x6F\x20\x77\x6F\x72\x6C\x64",
+                                   buffer,
+                                   length );
```
Comment (ActoryOu) [inline test/unit-test/rtp_api/rtp_api_utest.c]: Can you help move this expected result into a variable with comments? That would be easier to read.
For example:
```
uint8_t expectedSerializedBuffer = {
    0x80, ... /* RTP header */
    ... /* timestamp/ssrc/etc */
    ... /* RTP payload - "hello world" */
}

...

TEST_ASSERT_EQUAL_UINT8_ARRAY( expectedSerializedBuffer,
                                   buffer,
                                   length );
```

File: test/unit-test/rtp_api/rtp_api_utest.c [hunk 3]
```diff
@@ -0,0 +1,892 @@
...
+                            buffer,
+                            &length );
+
+    TEST_ASSERT_EQUAL( RTP_RESULT_OK,
+                       result );
+
+    TEST_ASSERT_EQUAL( 32,
+                       length );
+
+    TEST_ASSERT_EQUAL_UINT8_ARRAY( ( uint8_t * )"\x82\x60\x04\xD2\x12\x34\x56\x78\x87\x65\x43\x21\x11\x22\x33\x44\x55\x66\x77\x88\x68\x65\x6C\x6C\x6F\x20\x77\x6F\x72\x6C\x64",
```
Comment (ActoryOu) [inline test/unit-test/rtp_api/rtp_api_utest.c]: same here

File: test/unit-test/rtp_api/rtp_api_utest.c [hunk 4]
```diff
@@ -0,0 +1,892 @@
...
+                            buffer,
+                            &length );
+
+    TEST_ASSERT_EQUAL( RTP_RESULT_OK,
+                       result );
+
+    TEST_ASSERT_EQUAL( 36,
+                       length );
+
+    TEST_ASSERT_EQUAL_UINT8_ARRAY( ( uint8_t * )"\x90\x60\x04\xD2\x12\x34\x56\x78\x87\x65\x43\x21\x00\x00\x00\x02\x11\x22\x33\x44\x55\x66\x77\x88\x68\x65\x6C\x6C\x6F\x20\x77\x6F\x72\x6C\x64",
```
Comment (ActoryOu) [inline test/unit-test/rtp_api/rtp_api_utest.c]: same here

File: test/unit-test/rtp_api/rtp_api_utest.c [hunk 5]
```diff
@@ -0,0 +1,892 @@
...
+
+    result = Rtp_Serialize( &ctx,
+                            &packet,
+                            NULL,
+                            &length );
+
+    TEST_ASSERT_EQUAL( RTP_RESULT_OK,
+                       result );
+
+    TEST_ASSERT_EQUAL( 24,
```
Comment (ActoryOu) [inline test/unit-test/rtp_api/rtp_api_utest.c]: Declare a expectedLength with comment.
```
size_t expectedSerializedBufferLength = 24; /* RTP header ? bytes + payload 12 bytes. */
TEST_ASSERT_EQUAL( expectedSerializedBufferLength, length);
```

File: test/unit-test/rtp_api/rtp_api_utest.c [hunk 6]
```diff
@@ -0,0 +1,892 @@
...
+
+    TEST_ASSERT_EQUAL_HEX32( 0x55667788,
+                             packet.header.extension.pExtensionPayload[1] );
+}
+
+/*-----------------------------------------------------------*/
+/**
+ * @brief Validate Rtp_DeSerialize with a valid packet with padding.
+ */
+void test_Rtp_DeSerialize_Pass_WithPadding( void )
```
Comment (ActoryOu) [inline test/unit-test/rtp_api/rtp_api_utest.c]: Add some more padding test cases. Refer to #19.

File: test/unit-test/rtp_api/rtp_api_utest.c (hunk 1 above)
Comment (t2-g) [reply test/unit-test/rtp_api/rtp_api_utest.c]: I have used the local buffer only in my test cases, so removing the global buffer's declaration.

**Review by ActoryOu — DISMISSED** [review]

**Review by aggarg — APPROVED** [review]

**Review by ActoryOu — APPROVED** [review]

**Pull request merged**